tests/e2e/redis/parallel-ingest-redis-serving.py

```diff
@@ -0,0 +1,185 @@
+        )
+
+        cls.driver_entity = Entity(
+            name="driver_id",
+            description="Driver entity for car rides",
+            value_type=ValueType.STRING,
+            labels={"team": "matchmaking", "common_key": "common_val"},
+        )
+
+        cls.basic_ft_spec = FeatureTable.from_yaml(
+            f"{DIR_PATH}/specifications/dev_ft.yaml"
+        )
+
+    def test_discovery(self, client):
+
+        # ApplyEntity
+        client.apply_entity(self.customer_entity)
+        client.apply_entity(self.driver_entity)
+
+        # GetEntity Check
+        assert client.get_entity(name="customer_id") == self.customer_entity
+        assert client.get_entity(name="driver_id") == self.driver_entity
+
+        # ListEntities Check
+        common_filtering_labels = {"common_key": "common_val"}
+        matchmaking_filtering_labels = {"team": "matchmaking"}
+
+        actual_common_entities = client.list_entities(labels=common_filtering_labels)
+        actual_matchmaking_entities = client.list_entities(
+            labels=matchmaking_filtering_labels
+        )
+        assert len(actual_common_entities) == 2
+        assert len(actual_matchmaking_entities) == 1
+
+        # ApplyFeatureTable
+        client.apply_feature_table(self.basic_ft_spec, PROJECT_NAME)
+
+        # GetFeatureTable Check
+        actual_get_feature_table = client.get_feature_table(name="dev_featuretable")
+        assert actual_get_feature_table == self.basic_ft_spec
+
+        # ListFeatureTables Check
+        actual_list_feature_table = [
+            ft for ft in client.list_feature_tables() if ft.name == "dev_featuretable"
+        ][0]
+        assert actual_list_feature_table == self.basic_ft_spec
+
+    def test_basic_retrieval(self, client):
+        # TODO: Add ingest and retrieval check
+        pass
+
+
+@pytest.mark.incremental
+class TestAllTypesIngestionRetrieval:
+    def setup_class(cls):
+        prefix = "alltypes_ingestion"
+        suffix = str(int(datetime.now().timestamp()))
+        batch_source = DataSource(
+            type=SourceType(1).name,
+            field_mapping={
+                "ride_distance": "ride_distance",
+                "ride_duration": "ride_duration",
+            },
+            options=FileOptions(file_format="parquet", file_url="file://feast/*"),
+            timestamp_column="ts_col",
+            date_partition_column="date_partition_col",
+        )
+
+        cls.alltypes_entity = Entity(
+            name="alltypes_id",
+            description="Driver entity for car rides",
+            value_type=ValueType.STRING,
+            labels={"cat": "alltypes"},
+        )
+
+        cls.alltypes_ft_name = f"{prefix}_alltypes_{suffix}"
+        cls.alltypes_ft_spec = FeatureTable(
+            name="alltypes",
+            entities=["alltypes_id"],
+            features=[
+                Feature(name="float_feature", dtype=ValueType.FLOAT).to_proto(),
```

Comment:
Why do we have to convert to_proto()? are users expected to do this as well?

Reply:
Removed.